Google.Api.Gax.Grpc.IntegrationTests/ClientBuilderBaseTest.cs

```diff
@@ -340,7 +391,29 @@ private protected override ChannelBase CreateChannel(string endpoint, ChannelCre
                 return ChannelCreated;
             }
 
+            protected override GoogleCredential SetUseJwtWithScopesFlag(GoogleCredential credential)
```

Comment:
I would rather we don't know much about the credentials, than do something like this, wich requieres the method to be unnecessarily virtual.

You can check CanUseChannelPool instead.

Reply:
@jskeet how would you feel about adding in ClientBuilderBase:

```
internal GoogleCredential GetGoogleCredential() {...}
internal Task<GoogleCredential> GetGoogleCredentialAsync() {...}
```
so we can write tests for the GoogleCredential that's used. This would be done in the same spirit as GetChannelCredentials in that validation has already been done, so we don't care if the builder really can use a GoogleCredential or not, we just give the best one that we can?

Or maybe we can have those be `protected virtual` to allow other types of GoogleCredentials in descendants, for instance impersonate credentials which we currently don't accept?

Of course, all of this can be done in another PR, and in this one keep the checks to the minimum.

Reply:
Yes, I always forget about not exposing Google.Apis.Auth in Gax, hmmm, ...

OK, I can work on adding the internal methods next week or the other. @TimurSadykov, are you OK to test the bare minimum now, like CanUseChannelPool, etc. and then we can improve testing separately?

Reply:
@amanda-tarafa yes, that can work right now, I'll remove the SetUseJwtWithScopesFlag method